ThunderCloud/TSCBadgeController.h

```diff
@@ -21,41 +22,41 @@
  @abstract A `NSArray` of `TSCBadge`s
  @discussion Gets set when the `TSCBadgeController` is initalized
  */
-@property (nonatomic, strong) NSMutableArray *badges;
+@property (nonatomic, strong, nonnull) NSMutableArray<TSCBadge *> *badges;
 
 /**
  Returns a shared instance of `TSCBadgeController`
  */
-+ (instancetype)sharedController;
++ (nonnull instancetype)sharedController;
 
 /**
- Returns a `TSCBadge` for the given id
+ Returns a `TSCBadge` for the given id or nil if no badge is found
  @param badgeId The unique id for the `TSCBadge`
  */
-- (TSCBadge *)badgeForId:(NSString *)badgeId;
+- (nullable TSCBadge *)badgeForId:(nonnull NSString *)badgeId;
 
 /**
  Returns a BOOL for whether the user had unlocked the badge or not
  @param badgeId The unique id for the `TSCBadge`
  */
-- (BOOL)hasEarntBadgeWithId:(NSString *)badgeId;
+- (BOOL)hasEarntBadgeWithId:(nonnull NSString *)badgeId;
 
 /**
  Marks and saves a badge as earnt
  @param badgeId The unique id for the `TSCBadge`
  */
-- (void)markBadgeAsEarnt:(NSString *)badgeId;
+- (void)markBadgeAsEarnt:(nullable NSString *)badgeId;
```

Comment:
Shouldn't this be non-null? Is there a benefit of letting null be passed to mark a badge as earnt?